### PR TITLE
web: initial slot end to end support

### PR DESCRIPTION
**Pull request by yugangw-msft**
Ensure supports from create, link to git, swap, to deploy. Those are still the minimum, but enough to get basic functions.  This will get #1255 resolved.
In the next period, we should consider to 
1. enable private git repo.
2. create slot from existing configuration sources
3. extend `slot delete` to support auto delete site, etc

### CODE REVIEW

**Comment by yugangw-msft** [comment]
@derekbekoe @mayurid, this is low risk, and webapp specific change. Would you review it and consider included in the upcoming release?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -72,6 +72,56 @@
     short-summary: commands to configure hostnames
 """
 
+helps['appservice web log'] = """
+    type: group
+    short-summary: commands to manager logs
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: typo: manage

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -60,9 +60,12 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs): # pylin
 register_cli_argument('appservice web deployment user', 'password', help='password, will prompt if not specified')
 
 register_cli_argument('appservice web deployment slot', 'slot', help='the name of the slot')
-register_cli_argument('appservice web deployment slot', 'webapp', help='Name of the webapp')
-register_cli_argument('appservice web deployment slot', 'auto_swap_slot', help='target slot to swap', default='production')
+register_cli_argument('appservice web deployment slot', 'webapp', arg_type=name_arg_type, completer=get_resource_name_completion_list('Microsoft.Web/sites'),
+                      help='Name of the webapp', id_part='name')
+register_cli_argument('appservice web deployment slot', 'auto_swap_slot', help='target slot to auto swap', default='production')
 register_cli_argument('appservice web deployment slot', 'disable', help='disable auto swap', action='store_true')
+register_cli_argument('appservice web deployment slot', 'target_slot', help="target slot to swap, default to 'production'")
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: Can we not use `default='production'` here?

Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: `production` is not yet documented literal, so i would like avoid it

**Pull request merged**